reports/cost_diff.go

```diff
@@ -0,0 +1,119 @@
+//   Copyright 2018 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package reports
+
+import (
+	"context"
+	"database/sql"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/trackit/jsonlog"
+	"github.com/trackit/trackit-server/aws"
+	"github.com/trackit/trackit-server/costs/diff"
+)
+
+type costDiffValue struct {
+	Cost      string
+	Variation string
+}
+
+type costDiffProduct map[string]costDiffValue
+
+func isInList(dateList []string, date string) bool {
+	for _, item := range dateList {
+		if item == date {
+			return true
+		}
+	}
+	return false
+}
+
+func getDates(rawData map[string]costDiffProduct) (dateList []string) {
+	dateList = make([]string, 0)
+	for _, product := range rawData {
+		for date := range product {
+			if !isInList(dateList, date) {
+				dateList = append(dateList, date)
+			}
+		}
+	}
+	sort.Strings(dateList)
+	return
+}
+
+func formatCostDiff(data []diff.PricePoint) (values costDiffProduct) {
+	values = make(costDiffProduct)
+	for _, key := range data {
+		var date string
+		if splittedDate := strings.Split(key.Date, "T"); len(splittedDate) > 0 {
```

Comment:
Always returns `true`, so you can remove your `else` and your `if` statements.

Reply:
This statement prevent issues in case of bad formatted date. Without this if/else, if date is bad formatted initially, the cell will be empty. With if/else cell will contain the bad formatted date but it will not be empty wich is IMO a better choice...